packages/flutter/lib/src/material/material.dart

```diff
@@ -35,11 +97,24 @@ class Material extends StatelessComponent {
   final int elevation;
   final Color color;
   final TextStyle textStyle;
+  final CustomMaterialPainter painter;
+
+  static MaterialInkController of(BuildContext context) {
```

Comment:
`of()` functions return all kinds of random things currently:
* `DefaultTextStyle.of()` returns a `TextStyle`
* `DefaultAssetBundle.of()` returns an `AssetBundle`
* `IconTheme.of()` returns an `IconThemeData`
* `Theme.of()` returns a `ThemeData`
* `MediaQuery.of()` returns a `MediaQueryData`
* `ButtonTheme.of()` returns a `ButtonColor`
* `PageStorage.of()` returns a `PageStorageBucket`
* `ModalRoute.of()` returns a `ModalRoute`
* `Hero.of()` returns a `Map<Object, HeroHandle>`
* `Scaffold.of()` returns a `ScaffoldState`
* `Navigator.of()` returns a `NavigatorState`

I agree that it's a bit weird. Not sure what the right solution is though. Right now, `of()` returns "the thing you care about that is vaguely related to the class on which you call it". :-)